### PR TITLE
Provides documentation for M1 GA MVP release usage

### PR DESCRIPTION
**Pull request by diverdane**
### Desired Outcome

M1 Push to File features are documented in the README.md and PUSH_TO_FILE.md files
for the GA MVP release.

### Implemented Changes

M1 Push to File features are documented for GA MVP release.

### Connected Issue/Story

CyberArk internal issue link: ONYX-13494

### Definition of Done

- README.md updated
- PUSH_TO_FILE.md updated

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: PUSH_TO_FILE.md [hunk 1]
```diff
@@ -1,67 +1,276 @@
...
 
-## Prerequisites/Assumptions
-- This guide does not cover Conjur configuration and setup. Please refer to
-  [Secrets Provider for Kubernetes documentation](https://docs.conjur.org/Latest/en/Content/Integrations/k8s-ocp/cjr-secrets-provider-lp.htm) for more information.
-- Push to File feature requires the Secrets Provider must be run as an init container.
-- Configuration of the Secrets Provider must be done using Annotations rather than using
-  environment variables
-  This reference describes how to configure push to file using the Secrets Provider. 
+   - In this step, you define a Conjur host used to authenticate the
+     `cyberark-secrets-provider-for-k8s container` with the Kubernetes
```
Comment (john-odonnell) [inline PUSH_TO_FILE.md:98]: Nit:
```suggestion
     `cyberark-secrets-provider-for-k8s` container with the Kubernetes
```

File: PUSH_TO_FILE.md [hunk 2]
```diff
@@ -82,48 +291,60 @@ for a description of each environment variable setting:
...
 | `conjur.org/conjur-secrets.{secret-group}`      | Push to File config is not available with environmental variables | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
-| `conjur.org/secret-file-path.{secret-group}`    | Push to File config is not available with environmental variables | Path for secrets file to be written. <br> For the initial release of push-to-file the secret file path for the shared secrets must be '/conjur/secrets' . The file path must also include a file name (i.e. must not be a directory). Values ending with `/` are rejected and cause the Secrets Provider to abort.
+| `conjur.org/secret-file-path.{secret-group}`    | Push to File config is not available with environmental variables | Path for secrets file to be written. <br> The file path must include a file name (i.e. must not be a directory). Values ending with `/` are rejected and cause the Secrets Provider to abort.
```
Comment (john-odonnell) [inline PUSH_TO_FILE.md:294]: If these doc updates are for M1 GA, should they include the ability to specify only a directory, where SP will use the group name as a default filename? ([unit test for this case](https://github.com/cyberark/secrets-provider-for-k8s/blob/main/pkg/secrets/pushtofile/secret_group_test.go#L289)) I'm not sure if this is out of scope for the story you're working on.

**Review by john-odonnell — CHANGES_REQUESTED** [review]
Great content, reads really well. Just left a couple small comments.

File: PUSH_TO_FILE.md (hunk 2 above)
Comment (diverdane) [reply PUSH_TO_FILE.md]: Yeah, this is missing details. I've updated the description to reflect current behavior (i.e. a directory can be specified for file path if custom template is not being used).

File: PUSH_TO_FILE.md [hunk 3]
```diff
@@ -75,55 +294,71 @@ for a description of each environment variable setting:
...
+| `conjur.org/conjur-secrets.{secret-group}`      | Note\* | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
+| `conjur.org/secret-file-path.{secret-group}`    | Note\* | Relative path for secret file or directory to be written. This path is assumed to be relative to the respective mount path for the shared secrets volume for each container.<br><br>If the `conjur.org/secret-file-format.{secret-group}` is set, then this secret file path must also be set, and it must include a file name (i.e. must not end in `/`).<br><br>If the `conjur.org/secret-file-format.{secret-group}` is not set, then this secret file path defaults to `{secret-group}.{secret-group-file-format}`. For example, if the secret group name is `my-app`, and the secret file format is set for YAML, the the secret file path defaults to `my-app.yaml`.
```
Comment (szh) [inline PUSH_TO_FILE.md:304]: > If the `conjur.org/secret-file-format.{secret-group}` is set, then this secret file path must also be set, and it must include a file name (i.e. must not end in `/`).

This should be secret-file-template

**Review by szh — CHANGES_REQUESTED** [review]

File: PUSH_TO_FILE.md (hunk 3 above)
Comment (diverdane) [reply PUSH_TO_FILE.md:304]: Ah, good catch! That's a very confusing typo.

File: PUSH_TO_FILE.md [hunk 4]
```diff
@@ -75,55 +294,71 @@ for a description of each environment variable setting:
...
+| `conjur.org/conjur-secrets.{secret-group}`      | Note\* | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
+| `conjur.org/secret-file-path.{secret-group}`    | Note\* | Relative path for secret file or directory to be written. This path is assumed to be relative to the respective mount path for the shared secrets volume for each container.<br><br>If the `conjur.org/secret-file-template.{secret-group}` is set, then this secret file path must also be set, and it must include a file name (i.e. must not end in `/`).<br><br>If the `conjur.org/secret-file-format.{secret-group}` is not set, then this secret file path defaults to `{secret-group}.{secret-group-file-format}`. For example, if the secret group name is `my-app`, and the secret file format is set for YAML, the the secret file path defaults to `my-app.yaml`.
```
Comment (szh) [inline PUSH_TO_FILE.md:304]: Type here too:
> If the `conjur.org/secret-file-format.{secret-group}` is not set

should be secret-file-template as well

Comment (diverdane) [reply PUSH_TO_FILE.md]: D'oh! Thanks again!

**Review by szh — APPROVED** [review]
Looks great!

**Review by john-odonnell — APPROVED** [review]
LGTM 👍
